### DISCUSSION TITLE
Multiple Instances of 2 Ch without Building in Xcode?

### DISCUSSION THREAD

**Discussion by triwaves** [ExistentialAudio/BlackHole#472] [Q&A] [answered]
Hello,

I know the suggestion is to just build with Xcode to get unique IDs for each instance. I don't use a Mac a whole lot and don't have room to setup the whole xcode environment for one task. (not even sure it will work!)

Given it boils down to the ID of the instance is there some way to manually clone the driver and edit a configuration file or something to provide that different ID? Or edit something in a .dmg package? 

Would like to run 4x of the 2 Ch devices for my experiment. 

Thanks!

**Comment by devinroth** [comment, marked as answer]
Nope sorry you need to recompile.

If you want to sponsor the project as a Lunar Sponsor I’ll build and sign them for you.
https://github.com/sponsors/ExistentialAudio

Devin

**Reply by ghost** [reply]
We route OBS output (Speaker voice) to the Zoom microphone via BlackHole, but we'd also want to route Zoom output (audience comments) back to OBS for recordings (Speaker + audience interaction).
We have to try if this would work in theory, but if it does, we'd also be interested in running multiple BlackHole instances.
Because we often migrate machines, it would be nice if BlackHole could just be installed via brew, even for the multi-instance setup.

Would you still accept PRs to the install script (and, as importantly, the brew casks) which would allow building a `blackhole-2x2ch` cask, which would create `BlackHole 2ch A` and `BlackHole 2ch B` instances? From what I can tell, this mostly comes down to adding another loop around https://github.com/ExistentialAudio/BlackHole/blob/a6f82f4caab377af1441b5cf20f0adad4bcb493e/Installer/create_installer.sh#L7

Or are customized (such as multi-instance) builds the basis of funding for this project that you wouldn't be willing to give up?

**Reply by devinroth** [reply]
Honestly it largely comes down to time and unfortunately I hardly receive any money from sponsorships. 

Why not just fork BlackHole and do it yourself?

**Reply by tobe-chronleapLab** [reply]
> Nope sorry you need to recompile. If you want to sponsor the project as a Lunar Sponsor I’ll build and sign them for you. https://github.com/sponsors/ExistentialAudio Devin
> […](#)
@devinroth i'm a Lunar Sponsor. How do i reach out to you to help with the building and signing? 
Thanks.

**Comment by triwaves** [comment]
Thanks for quick response Devin , but Lunar is a bit out of price range as I'm just conducting an experiment as a part of a larger solution. Your component has promise but I don't know enough at this point...   maybe I'll just have to free up some drive space if it looks like we really need the 4x to do an assessment. 

Thanks

**Comment by devinroth** [comment]
Thanks for the sponsorship! Send me an email at ***@***.*** with details.

Devin
